cpp/include/nvstrings/NVText.h

```diff
@@ -143,6 +143,15 @@ class NVText
      */
     static unsigned int edit_distance( distance_type algo, NVStrings& strs1, NVStrings& strs2, unsigned int* results, bool devmem=true );
 
+    /**
+     * @brief Compute the edit distance between each pair of strings in given nvstrings object
+     * @param algo The edit distance algorithm to use for the computation.
+     * @param strs Strings to process.
+     * @param devmem True if results in device memory.
+     * @return 0 if successful.
```

Comment:
Don't return 0 if successful. Return the results. Throw an exeption if there is a failure.